### DISCUSSION TITLE
Migrate from ZHA

### DISCUSSION THREAD

**Discussion by adi90x** [Koenkk/zigbee2mqtt#16739] [General]
Hello, 

Just want to move from ZHA integration to Z2M, I have a backup of ZHA with all keys and config.

Tried to set pan_id , network_keys and then some devices with a friendly name and none seems to show.

Is there a way to get those without re-pairing everything ( especially switch that are behind some walls :) ?   

Regards,
Adrien

**Comment by Koenkk** [comment]
The only way to migrate from zha -> z2m is to re-pair everything. 

I recommend to:
- Use a different pan_id and network_key
- Use this chance to also find an optimal zigbee channel (since that requires re-pairing) https://www.zigbee2mqtt.io/advanced/zigbee/02_improve_network_range_and_stability.html#adding-routers-to-your-network
- Power off all Zigbee devices
- Start powering them on one-by-one and pair.

**Reply by alexdelprete** [reply]
> The only way to migrate from zha -> z2m is to re-pair everything.

I was exploring the possibility to migrate from zha to z2m, without changing coordinator (I use a Tube's efr32_mgm12_pro1).

I imagined that if I didn't change coordinator, I would not need to re-pair everything. Can you explain technically why re-pair is needed? Wouldn't devices be re-discovered?

Thanks for any clarification about this.

**Comment by adi90x** [comment]
Hello,

My question was from zha -> z2m, does your reply still applied ?

Regards,
Adrien

Le dim. 19 févr. 2023, 10:19, Koen Kanters ***@***.***> a
écrit :

**Reply by Koenkk** [reply]
Updated, my comment, I meant zha -> z2m.

**Reply by adi90x** [reply]
Thanks a lot, will assess opportunity to switch ( and repair all switch that are behind walls +- 20 behind walls and +-80 in total to repair...

**Reply by adi90x** [reply]
I haven't check it for the moment but there is no way to manually add device into z2m database ?

**Comment by Hedda** [comment]
FYI, for reference also see this related discussion about migrating from one radio adapter type to any other radio adapter type -> https://github.com/Koenkk/zigbee2mqtt/discussions/16478

Regardless how you proceed to recommend that you first perform a backup of your network configuration to file using "zigpy-cl"i as a tool:

https://github.com/zigpy/zigpy-cli#network-backup (suggest also read https://github.com/zigpy/open-coordinator-backup)

> The only way to migrate from zha -> z2m is to re-pair everything.

@Koenkk There is a somewhat related discussion about ZHA to Zigbee2MQTTT migrations in the Home Assistant community forum here:

https://community.home-assistant.io/t/zha-to-zigbee2mqttt-migration-using-backup-file/473687

Technically it is possible to migrate  Zigbee network without re-joining all devices, but still need to configure them in app, see example Home Assistant has guides for the reverse, with migration from Zigbee2MQTT to ZHA or from deCONZ to ZHA:

https://skyconnect.home-assistant.io/migrate-zigbee-zha/

https://skyconnect.home-assistant.io/migrate-conbee/

Sure, seamless migration between different Zigbee gateway applications is not really possible since most information is stored in the application and not on the Zigbee adapter, however, if you are moving your Zigbee Coordinator then should, in theory, be possible to not have to keep your network setting by not forming a new Zigbee network instead manually using advanced network config in Zigbee2MQTT by typing your same old PAN ID and network encryption key (Zigbee Transport Key) that you used in the ZHA integration, as that way your devices should at least automatically join Zigbee2MQTT your new Zigbee gateway application, but as far as Zigbee2MQTT is concerned they will be new devices so you will not keep any device configurations. 

You could kind of compare that to using the same SSID (Network Name) and WPA2 key (Password) for your new Wi-Fi Router/AP as your old Wi-Fi Router/AP. Zigbee devices should join a network if PAN ID, Zigbee Transport Key, and Zigbee channel are the same as old setup.

https://www.zigbee2mqtt.io/guide/configuration/zigbee-network.html

https://www.zigbee2mqtt.io/advanced/zigbee/03_secure_network.html
